core/storage/classifier/gae_models.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2016 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Models for storing the classification models."""
+
+from core.platform import models
+import utils
+
+from google.appengine.ext import ndb
+
+(base_models,) = models.Registry.import_models([models.NAMES.base_model])
+
+
+class ClassifierModel(base_models.BaseModel):
+    """Storage model for classifier used for answer classification.
+
+    The id/key of instances of this class has the form
+    {{exp_id}}.{{random_hash_of_16_chars}}..
+    """
+
+    # The exploration_id of the exploration to whose state the model belongs.
+    exp_id = ndb.StringProperty(required=True)
+    # The exploration version when the model was created.
+    exp_version_when_created = ndb.IntegerProperty(required=True)
+    # The name of the state to which the model belongs.
+    state_name = ndb.StringProperty(required=True)
+    # The ID of the algorithm used to create the model.
+    algorithm_id = ndb.IntegerProperty(required=True)
+    # The actual model used for classification. Immutable, unless a schema
+    # upgrade takes place.
+    cached_classifier_data = ndb.JsonProperty(required=True)
+    # The schema version for the data that is being classified
+    data_schema_version = ndb.IntegerProperty(required=True, default=1)
```

Comment:
@seanlip the current schema can be different for different type of classifiers, I guess. For now, there is just one classifier but let's say we have another classifier tomorrow which expects data in a slightly different format (perhaps a code classifier). In this case, we can't have a CURRENT_CLASSIFIER_DATA_SCHEMA_VERSION because it'll something for classifier A and something else for classifier B.